### PR TITLE
Cypress full tests from main

### PR DESCRIPTION
**Pull request by danlester**
Cypress tests in tests.yaml workflow for Minikube. Tests login only.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -154,7 +153,34 @@ jobs:
           nslookup github-actions.qhub.dev
       - name: Curl jupyterhub login page
         run: |
-          curl -k https://github-actions.qhub.dev/hub/home -i
+          curl -k --retry 5 --retry-connrefused https://github-actions.qhub.dev/hub/login
```
Comment (costrouc) [inline .github/workflows/test.yaml:156]: `curl -k https://github-actions.qhub.dev/hub/home -i` should now work. Retry is no longer needed

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -7,6 +7,7 @@ on:
   push:
     branches:
       - main
+      - cypress-main
```
Comment (costrouc) [inline .github/workflows/test.yaml:10]: Assume this was only for testing? And can be removed?

**Review by costrouc — COMMENTED** [review]
Just those two changes and after that looks good to me. Looking forward to having the web UI testing.

**Comment by danlester** [comment]
Great, thanks for your comments - have accepted all and I will just merge.

**Pull request merged**